pkg/apis/velero/v1/download_request_types.go

```diff
@@ -29,13 +29,15 @@ type DownloadRequestSpec struct {
 type DownloadTargetKind string
```

Comment:
The `+kubebuilder:validation:Enum` line above needs the new entries added. Did you manually add those to the downloadrequest CRD file? CRDs are generated via `make update`, so anything added manually will be removed on next update.

Reply:
I had manually edited it earlier, incorporated your comment and added it to the annotation. Ran the autogen

Let me know if any other concerns with this PR

I will also add test cases if rest of the things look good to you.